python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -368,6 +414,18 @@ def speculative_generate(self,
     e2e_tic = 0.0
 
     self.clear_benchmarks()
+    attention_mask = model_kwargs["attention_mask"]
+
+    if len(input_ids.shape) == 1:
+        batch_size = 1
+    else:
+        batch_size = input_ids.size(0)
+    pad_token_id = generation_config.pad_token_id
```

Comment:
When bs==1, we don't need padding and mask. I think we should separated bs==1 and bs!=1.